inductiva/_benchmarks/benchmark.py

```diff
@@ -0,0 +1,119 @@
+from inductiva.projects.project import Project
+from inductiva.resources import machine_groups
+
+
+class Benchmark(Project):
+    """Represents the benchmark runner."""
+
+    def __init__(self, name: str, append: bool = True):
+        """
+        Initializes a new Benchmark instance.
+
+        Args:
+            name (str): The name of the benchmark runner.
+            append (bool): Indicates whether to allow adding runs to the 
+            existing benchmark (default is True).
+        """
+        super().__init__(name=name, append=append)
+        self.runs = []
+
+    def add_run(
+        self,
+        simulator: Simulator,
+        input_dir: str,
+        on: types.ComputationalResources,
+        **kwargs,
+    ) -> Self:
+        """
+        Adds a simulation run to the benchmark.
+
+        Args:
+            simulator (Simulator): The simulator to be used for this run.
+            input_dir (str): The directory containing input files for the
+            simulation.
+            on (types.ComputationalResources): The computational resources to
+            run the simulation on.
+            **kwargs: Additional keyword arguments for the simulator run.
+
+        Returns:
+            Self: The current instance for method chaining.
+        """
+        self.runs.append((simulator, input_dir, on, kwargs))
+        return self
+
+    def run(self, num_repeats: int = 2) -> Self:
+        """
+        Executes all added runs.
+
+        Each run is executed the specified number of times, and the collection
+        of runs is cleared afterwards.
+
+        Args:
+            num_repeats (int): The number of times to repeat each simulation
+            run (default is 2).
+
+        Returns:
+            Self: The current instance for method chaining.
+        """
+        with self:
+            for simulator, input_dir, machine_group, kwargs in self.runs:
+                machine_group.start(wait_on_pending_quota=True)
+                for _ in range(num_repeats):
+                    simulator.run(input_dir=input_dir,
+                                  on=machine_group,
+                                  **kwargs)
+            self.runs.clear()
+        return self
+
+    def wait(self) -> Self:
+        """
+        Waits for all running tasks to complete.
+
+        Returns:
+            Self: The current instance for method chaining.
+        """
+        tasks = self.get_tasks()
+        for task in tasks:
+            task.wait(download_std_on_completion=False)
+        return self
+
+    def to_dict(self):
+        """
+        Compiles results from all completed tasks into a dictionary.
+
+        Returns:
+            dict: A dictionary containing the results of each task, organized 
+            by virtual machine type.
+        """
+        results = {}
+        tasks = self.get_tasks()
+        for task in tasks:
+            task_info = task.info
+            tmp1, tmp2, *_ = task_info.executer.vm_name.split("-")
```

Comment:
I think that here there is some hardcoded assumption that will only work for GCP vm names -- can we encapsulate this into some function? And see if what we need to extract also works for other machine groups? We will want to use this same Benchmark class to also include machine groups of local machines, for example...

Reply:
This was removed in the most recent PR related to benchmarks, which focused on exporting and gathering benchmarking metrics and data.